src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,215 @@
+    @Override
+    public Car create(Car car) {
+        String query = "INSERT INTO cars (model, manufacturer_id) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query,
+                        Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, car.getModel());
+            statement.setObject(2, car.getManufacturer().getId());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                car.setId(resultSet.getObject(1, Long.class));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't create "
+                    + car + ". ", e);
+        }
+        car.setDrivers(getAllByCar(car.getId()));
+        return car;
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        String query = "SELECT cars.id AS car_id, cars.model AS model, "
+                + "manufacturers.id AS manufacturer_id, "
+                + "manufacturers.country AS country, manufacturers.name AS manufacturer_name "
+                + "FROM cars INNER JOIN manufacturers "
+                + "ON manufacturers.id = cars.manufacturer_id "
+                + "WHERE cars.id = ? AND cars.is_deleted = FALSE";
+        Car car = new Car();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                car = setCar(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get car by id " + id, e);
+        }
+        car.setDrivers(getAllByCar(id));
+        return Optional.ofNullable(car);
+    }
+
+    @Override
+    public List<Car> getAll() {
+        List<Car> cars = new ArrayList<>();
+        String query = "SELECT cars.id AS car_id, cars.model AS model, "
+                + "manufacturers.id AS manufacturer_id, "
+                + "manufacturers.country AS country, manufacturers.name manufacturer_name "
+                + "FROM cars INNER JOIN manufacturers "
+                + "ON manufacturers.id = cars.manufacturer_id "
+                + "WHERE cars.is_deleted = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(setCar(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't get all cars from DB.", e);
+        }
+        for (int i = 0; i < cars.size(); i++) {
```

Comment:
```suggestion
        for (Car car: cars) {
```

мені здається цей фигляд циклу буде тут більш симпатично виглядати

Reply:
Done